src/main/java/mate/academy/spring/controller/OrderController.java

```diff
@@ -34,14 +34,20 @@ public OrderController(ShoppingCartService shoppingCartService,
     }
 
     @PostMapping("/complete")
-    public OrderResponseDto completeOrder(@RequestParam Long userId) {
-        ShoppingCart cart = shoppingCartService.getByUser(userService.get(userId));
+    public OrderResponseDto completeOrder(Authentication authentication) {
+        ShoppingCart cart = shoppingCartService.getByUser(
+                userService.findByEmail(authentication.getName()).orElseThrow(
+                        () -> new RuntimeException("User is not found.")
```

Comment:
```suggestion
                        () -> new RuntimeException("User  is not found. Email: " + email)
```
the same for other similar cases